02-r-programming.Rmd

```diff
@@ -187,13 +201,21 @@ summary(everything)
 
 ## Functions
 
-You only need to understand the very basics of functions for this exam.  Still, understanding functions helps you to understand *everything* in R, since R is a functional [programming language](http://adv-r.had.co.nz/Functional-programming.html), unlike Python, C, VBA, Java which are all object-oriented, or SQL which isn't really a language but a series of set-operations.
+You only need to understand the very basics of functions for this exam.  Still,
+understanding functions helps you to understand *everything* in R, since R is a 
+functional [programming language](http://adv-r.had.co.nz/Functional-programming.html), 
+unlike Python, C, VBA, Java which are all object-oriented, or SQL which isn't 
+really a language but a series of set-operations.
 
-Functions do things.  The convention is to name a function as a verb.  The function `make_rainbows()` would create a rainbow.  The function `summarise_vectors` would summarise vectors.  Functions may or may not have an input and output.  
+Functions do things.  The convention is to name a function as a verb.  The function
+`make_rainbows()` would create a rainbow.  The function `summarise_vectors()` 
+would summarise vectors.  Functions may or may not have an input and output.  
 
-If you need to do something in R, there is a high probability that someone has already written a function to do it.  That being said, creating simple functions is quite useful.
+If you need to do something in R, there is a high probability that someone has 
+already written a function to do it.  That being said, creating simple functions 
+is quite useful.
 
-**A function that does not return anything**
```

Comment:
This function actually returns, because `print()` returns its argument. In fact, all functions in R return something, though the returned value may be invisible.